controllers/rhmi/rhmi_controller.go

```diff
@@ -979,10 +981,10 @@ func (r *RHMIReconciler) preflightChecks(installation *rhmiv1alpha1.RHMI, instal
 		err := r.Status().Update(context.TODO(), installation)
 		if err != nil {
 			log.Infof("error updating status", l.Fields{"error": err.Error()})
-			return result, err
+			return err
 		}
 		log.Warning("preflight checks failed on useClusterStorage value")
-		return result, nil
+		return nil
```

Comment:
I think in here, we should be returning custom error. Reason being is that we do not want to continue if useClusterStorage is set neither to false or true...

Reply:
I dont think the reconcile will generally continue at the moment even without the custom error due to the following :
https://github.com/integr8ly/integreatly-operator/blob/edbe2c72475b1b74ad724685fa540aabc301d366/controllers/rhmi/rhmi_controller.go#L979-L980

https://github.com/integr8ly/integreatly-operator/blob/edbe2c72475b1b74ad724685fa540aabc301d366/controllers/rhmi/rhmi_controller.go#L372-L379

Though the logic probably should indicate an err rather than returning nil as you suggest 🤔

Reply:
By saying reconcile will continue I mean it would return to rhmi controller and continue regardless of what's the story with preflight state, but Anna pushed other changes meanwhile so it's not relevant, but I think it would be better to return an error indicating what's going on rather than nil. 
Generally, I think it's better to be as informative as we can so if the failure is caused by an error, let's return an error.